backend/controllers/authController.js

```diff
@@ -225,4 +225,42 @@ const resetPassword = async(req, res) => {
     }
 };
 
-export { register, verifyEmail, login, forgotPassword, verifyResetCode, resetPassword };
\ No newline at end of file
+// Get Profile Handler
+const getProfile = async (req, res) => {
+    try {
+        const user = await UserModel.findById(req.user.id).select('-password'); // Exclude password
+        if (!user) {
+            return res.status(404).json({ msg: 'User not found' });
+        }
+        return res.status(200).json(user);
+    } catch (err) {
+        console.error(`Error fetching user profile: ${err.message}`);
+        return res.status(500).json({ msg: 'Server error' });
+    }
+};
+
+// Update Profile Handler
+const updateProfile = async (req, res) => {
+    try {
+        const updatedData = req.body;
+        // Only allow certain fields to be updated
+        const allowedUpdates = ['first_name', 'last_name', 'phone', 'isDonor', 'isStudent'];
+        const updates = {};
+        for (let key of allowedUpdates) {
+            if (updatedData[key] !== undefined) {
+                updates[key] = updatedData[key];
+            }
+        }
```

Comment:
This is smart. I assume it's so that a user can update any feature without necessarily updating others right? I think Mongoose's findByIdAndUpdate automatically does that when you send any object. I'm not sure about this though but if it does, then we might not need an extra loop

Reply:
You are correct. The `allowedUpdates` check is important because it adds an explicit layer of validation and security.